### ISSUE TITLE
error thrown on mlabeledtr with mglyph

### ISSUE THREAD

**Issue by pkra** [zorkow/speech-rule-engine#279]
Example:

````js
const sre = require('speech-rule-engine');
sre.setupEngine({
  domain: 'mathspeak',
  style: 'default',
  locale: 'en',
  speech: 'deep',
  structure: true,
  mode: 'sync'
});
sre.engineReady();

const mml = '<math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><mtable displaystyle="true"><mlabeledtr><mtd id="mjx-eqn-3.1.10"><mtext>(3.1.10)</mtext></mtd><mtd><mrow class="MJX-TeXAtom-VCENTER"><mglyph src="Images/img2354b563c08bcdd40777a6bbee95ac36.svg" width="148pt" height="16pt"></mglyph></mrow><mrow id="texmlid28" /></mtd></mlabeledtr></mtable></math>'

console.log(sre.toEnriched(mml);
// TypeError: Cannot read property 'personality' of undefined 
//    at sre.SpeechRuleEngine.addPersonality_ (/tmp/sre/node_modules/speech-rule-engine/lib/sre.js:1128:391)
````

**Comment by pkra** [comment]
Reproduced on 2.3.0, 2.3.1.

**Comment by zorkow** [comment]
That's certainly linked to the non-treatment of `mglpyh` elements (see #165). Can you send me the svg so I can play with the expression.

**Comment by pkra** [comment]
This is a regression. It works on 2.2.2. I'll see about the glyph but it shouldn't matter, no?

**Comment by pkra** [comment]
It's more general, affecting tables without labels and mlgyphs. Let me try to shrink an example down.

**Comment by pkra** [comment]
Smaller example:

```xml
<math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><mtable columnalign="right left right left right left right left right left right left" rowspacing="3pt" columnspacing="0em 2em 0em 2em 0em 2em 0em 2em 0em 2em 0em" displaystyle="true"><mtr><mtd /><mtd><mi></mi><mspace width="2em" /><mo>+</mo><mrow class="MJX-TeXAtom-ORD"><mi mathvariant="double-struck">E</mi></mrow><mrow><mo>[</mo><mrow class="MJX-TeXAtom-ORD"><msubsup><mo>&#x222B;<!-- ∫ --></mo><mn>0</mn><mi>T</mi></msubsup><msup><mrow><mo>(</mo><msubsup><mrow><mo symmetric="true">&#x2016;</mo><mrow class="MJX-TeXAtom-ORD"><mi>u</mi><mo stretchy="false">(</mo><mo>&#x22C5;<!-- ⋅ --></mo><mo>,</mo><mi>s</mi><mo stretchy="false">)</mo></mrow><mo symmetric="true">&#x2016;</mo></mrow><mrow class="MJX-TeXAtom-ORD"><msup><mi>L</mi><mn>8</mn></msup><mo stretchy="false">(</mo><mrow class="MJX-TeXAtom-ORD"><mrow class="MJX-TeXAtom-ORD"><mi class="MJX-tex-caligraphic" mathvariant="script">O</mi></mrow></mrow><mo stretchy="false">)</mo></mrow><mn>8</mn></msubsup><mo>)</mo></mrow><mrow class="MJX-TeXAtom-ORD"><mi>p</mi><mo>&#x2212;<!-- − --></mo><mn>1</mn></mrow></msup></mrow><mo>]</mo></mrow><mo>,</mo><mspace width="1em" /><mtext>&#xA0;</mtext><mtext>&#xA0;</mtext></mtd></mtr></mtable></math>
```

even smaller, with a different error:

```
<math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><mtable columnalign="right left right left right left right left right left right left" rowspacing="3pt" columnspacing="0em 2em 0em 2em 0em 2em 0em 2em 0em 2em 0em" displaystyle="true"><mtr><mtd /><mtd><mi></mi><mspace width="2em" /><mo>+</mo><mrow class="MJX-TeXAtom-ORD"><mi mathvariant="double-struck">E</mi></mrow><mrow><mo>[</mo><mrow class="MJX-TeXAtom-ORD"><msubsup><mo>&#x222B;<!-- ∫ --></mo><mn>0</mn><mi>T</mi></msubsup></mrow><mo>]</mo></mrow><mo>&#x2264;<!-- ≤ --></mo><msub><mi>C</mi><mrow class="MJX-TeXAtom-ORD"><mi>T</mi><mo>,</mo><mi>p</mi></mrow></msub><mo>,</mo><mspace width="1em" /><mtext>&#xA0;</mtext><mtext>&#xA0;</mtext></mtd></mtr></mtable></math>
```

gives 

/tmp/sre/node_modules/xmldom-sre/dom.js:1283
                                return buf.join('');
                                           ^

RangeError: Maximum call stack size exceeded
    at Array.join (native)

**Comment by zorkow** [comment]
It's a regression in Mathspeak. It works fine with the old ChromeVox rules.

**Comment by zorkow** [comment]
We conflate two issues here. The first is a lack of treatment from the `mglyph`. 
The other two problems come from the non-breaking spaces.
```XML
<mtext>&#xA0;</mtext><mtext>&#xA0;</mtext>
```
Removing these removes the problem. I'll open a separate issue for those.

**Comment by pkra** [comment]
Thanks for looking into it quickly. I'll schedule some time to run 2.3.2 or above through 500K samples if you like.

**Comment by zorkow** [comment]
This was caused by new treatment of text and none-breaking spaces, resulting in multiple nodes, which still leads to empty speech.   
The position of the bug was actually flagged for review during beta. It appears no-one has ever used the 2.3.0-beta.0.

**Comment by zorkow** [comment]
Published 2.3.2
Unpublished 2.3.1 as there was some code missing.
